Technical/Performance-Benchmarking/performance-benchmarking-of-arangodb-single-instance-vs-split-databases-vs-clustering.md

```diff
@@ -0,0 +1,2030 @@
+| 083@1.0.0 | 3.304015 | 637.017334 | 12.434935|
+| 084@1.0.0 | 3.389799 | 635.764099 | 12.602417|
+
+<span style="color:blue;;">**Typologies**</span>
+
+| Typology ID | mn       | mx       | ave      |
+|-------------|----------|----------|----------|
+| 028@1.0.0   | 0.079802 | 3.140593 | 0.139582 |
+| 037@1.0.0   | 0.080601 | 3.142096 | 0.139708 |
+| 044@1.0.0   | 0.070701 | 3.303908 | 0.122282 |
+| 047@1.0.0   | 0.062301 | 3.698223 | 0.100181 |
+| 095@1.0.0   | 0.041901 | 3.031201 | 0.075914 |
+| 185@1.0.0   | 0.047101 | 3.589418 | 0.090269 |
+
+<span style="color:green; font-size: 24px;">****Analysis****</span>
+
+In this comprehensive analysis, which is based on examining the throughput of each configuration, as indicated by the FTPS metric. The analysis includes an evaluation of how effectively the configurations handle load, their responsiveness under stress, and their resilience in terms of fault tolerance and failover capabilities.
+
+- **Single Instance:** With an **FTPS** of 2240, this configuration demonstrated reliable performance for scenarios not requiring high availability or distributed processing. It's particularly suited for development, testing, or smaller-scale production environments.
+- **2 Split Databases:** The **FTPS** of 2304.5 suggests a robust capability to handle more substantial workloads, especially read-heavy ones, while offering some level of scalability without the full complexity of clustering.
+- **4 Split Databases:** The **FTPS** of 2496.83 indicates a strong ability to manage larger workloads, particularly those that are read-intensive. It also provides a degree of scalability without the need for the complete complexity of clustering.
+- **Clustering:** Despite an **FTPS** of 264.5, the clustering configuration shines in high availability and horizontal scalability. It's built to endure high-load scenarios and provides the foundation for a resilient, enterprise-level deployment.
+
+****Performance vs. Cost:****
+
+The evaluation of performance against cost reveals an interesting dynamic between the configurations. The **Single Instance** configuration proved to be the most cost-effective, nearly matching the highest FTPS achieved by the Split Databases setup, which boasted the top performance. This demonstrates a high return on investment for the Single Instance in terms of performance per dollar. On the other end, the Clustering configuration, while being the most expensive and the slowest, is valued for its superior resilience and fault tolerance, features that are often required for mission-critical applications.
+
+When considering the non-linear relationship between performance and cost, the Single Instance stands out for low- to medium-load scenarios where cost savings are paramount. In contrast, the additional expense of the Clustering setup is justified by its advanced capabilities, which are crucial for high-availability requirements despite its lower FTPS.
```

Comment:
Are we saying it is indeed possible to ramp up a clustered setup to 2500FTPS, it'll just cost a lot more?

Reply:
Correct , you can add way more vms (like 50 times due to how slow it is) for the clustered version to get it to that number but it would be better to change certain optimizations before even trying to attempt that.

Heres a document : arango-clustering-discussion-outcomes in the same folder